pkg/webhook/controlplane/ensurer.go

```diff
@@ -149,6 +148,133 @@ func keepNodeNetworkEnvVarIfPresentInOldDeployment(new, old *appsv1.Deployment,
 	}
 }
 
+// EnsureAdditionalProvisionUnits ensures that additional required system units are added, that are required during provisioning.
+func (e *ensurer) EnsureAdditionalProvisionUnits(ctx context.Context, gctx gcontext.GardenContext, new, _ *[]extensionsv1alpha1.Unit) error {
+
+	cluster, err := gctx.GetCluster(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, worker := range cluster.Shoot.Spec.Provider.Workers {
+		if worker.DataVolumes != nil {
+			switch worker.Machine.Image.Name {
+			case "flatcar":
+				extensionswebhook.AppendUniqueUnit(new, extensionsv1alpha1.Unit{
```

Comment:
that means that we cannot mix different worker types?
We only check if there is at least one worker of type Flatcar and then we add it?

If so it would be simpler/more readable to helper function that checks this (e.g. `hasFlatcarWorker`) and apply the unit once.
This could also be reused in both functions

Reply:
I'd prefer to do that on a per node basis, as now we have to assume that the whole cluster is homogenous. The issue is, that we don't have access to information to anything on the individual node basis, but just cluster scope.

Reply:
I'm fine with this, but then we should have some safety mechanisms:
- document that we only support worker pools with the same os. or at least if one uses flatcar
- have a validating webhook, or error check if this rule is broken